stingray/utils.py

```diff
@@ -163,67 +165,59 @@ def rebin_data(x, y, dx_new, yerr=None, method='sum', dx=None):
 
     step_size: float
         The size of the binning step
-
-    Examples
-    --------
-    >>> x = np.arange(0, 100, 0.01)
-    >>> y = np.ones(x.size)
-    >>> yerr = np.ones(x.size)
-    >>> xbin, ybin, ybinerr, step_size = rebin_data(
-    ...     x, y, 4, yerr=yerr, method='sum')
-    >>> np.allclose(ybin, 400)
-    True
-    >>> np.allclose(ybinerr, 20)
-    True
-    >>> xbin, ybin, ybinerr, step_size = rebin_data(
-    ...     x, y, 4, yerr=yerr, method='mean')
-    >>> np.allclose(ybin, 1)
-    True
-    >>> np.allclose(ybinerr, 0.05)
-    True
     """
 
     y = np.asarray(y)
     yerr = np.asarray(apply_function_if_none(yerr, y, np.zeros_like))
 
-    dx_old = apply_function_if_none(dx, np.diff(x), np.median)
+    if not dx:
+        dx_old = np.diff(x)
+    else:
+        dx_old = np.array([dx])
```

Comment:
So, it's not clear to me what `dx` stands for. This is the old resolution of data, but can't it be variable?

Reply:
Yes, it can, but I'm just not assuming it's a numpy array. I'll code in a special case for when it's a single number versus when it's not